src/blueprint/from_config.rs

```diff
@@ -358,8 +373,41 @@ fn update_modify(
     None => Valid::Ok(Some(b_field)),
   }
 }
+fn update_const_field(field: &config::Field, mut b_field: FieldDefinition, config: &Config) -> Valid<FieldDefinition> {
+  match field.const_field.as_ref() {
+    Some(const_field) => {
+      let mut data = const_field.data.to_owned();
+      // Add {key: null} pair if key is nullable and not provided in const value
+      if let Some(typ) = config.find_type(&field.type_of) {
+        if let Value::Object(ref mut map) = data {
+          for key in typ.fields.keys() {
+            if let Some(field) = typ.fields.get(key) {
+              let json_has_key = map.contains_key(key);
+              let field_required = field.required;
+              if !json_has_key && !field_required {
+                map.insert(key.to_owned(), Value::Null);
+              }
+            }
+          }
+        }
+      }
```

Comment:
Two questions 
- Why do we need this? If the value isn't available, sending `null` or not having the key itself is morally equivalent.
- Wouldn't we have to do this recursively?

Reply:
The code handles the following remark, https://github.com/tailcallhq/tailcall/pull/408#discussion_r1343552654. 

1. I think `null` is required. Here is the example,
```
type User {
    name: String
    age: Int
}

type Query {
    user: User @const(data: {name: "John"})
}
```
And query is asking for `age`,
```
user {
  age
  name
}
```
There could be better way to do this as I understood the original remark, it make sense. 

2. Now I think about it, I'll need to check. We currently have flat JSON so will add test for object containing object. But will wait for your reponse on the first answer/explanation. 

@tusharmath